src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java

```diff
@@ -113,6 +118,23 @@ private static FeedSource createFeedSource(Request req, Response res) throws IOE
         }
     }
 
+    private static boolean validate(Request req, FeedSource feedSource) {
+        if (feedSource.name == null || feedSource.name.isEmpty()) {
+            logMessageAndHalt(req, HttpStatus.BAD_REQUEST_400, "Feed source name must be provided");
+            return false;
+        }
+        List<FeedSource.FeedRetrievalMethod> retrievalMethods = feedSource.transformRules.stream()
+            .map(rule -> rule.retrievalMethods)
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList());
+        Set<FeedSource.FeedRetrievalMethod> retrievalMethodSet = new HashSet<>(retrievalMethods);
+        if (retrievalMethods.size() > retrievalMethodSet.size()) {
```

Comment:
Could these two lines be merged into this?
```java
.collect(Collectors.toSet());
```

Reply:
No, I don't think so. I want to explicitly check that the list is not larger than the set (i.e., that there are no duplicates). I'll add a comment for clarity.